### PR TITLE
WIP: redirect middleware

### PR DESCRIPTION
**Pull request by laq**
First setup of the backbone for the redirect middleware, this is missing discussion to make it a full feature.

### CODE REVIEW

File: common/djangoapps/edunext/models.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+from django.db import models
+from django.utils.text import compress_string
+
+from config_models.models import ConfigurationModel
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+class RateLimitConfiguration(ConfigurationModel):
```
Comment (felipemontoya) [inline common/djangoapps/edunext/models.py]: Este nombre que onda?

File: common/djangoapps/edunext/models.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+"""Models for the util app. """
+import cStringIO
```
Comment (felipemontoya) [inline common/djangoapps/edunext/models.py]: could we purge the imports?

File: common/djangoapps/edunext/middleware.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
+"""
+Middleware for Language Preferences
```
Comment (felipemontoya) [inline common/djangoapps/edunext/middleware.py]: Eventually we'll get to docstring this better

File: common/djangoapps/edunext/middleware.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    """
+    Middleware for Redirecting microsites
+    """
+
+    def process_request(self, request):
+        """
+        Redirect if domain in redirect list
+        """
+        # TODO discuss how and from where this redirect should work, new table? microsites table? sites table?
+        return HttpResponseRedirect('http://www.edunext.co')
```
Comment (felipemontoya) [inline common/djangoapps/edunext/middleware.py]: just leaving this as an open discussion, should we use `//` to let the protocol be automatic

**Comment by felipemontoya** [comment]
I have updated the feature, tested it on a stage server and added a 5 min cache to the db query

**Comment by felipemontoya** [comment]
@laq any thoughts?

**Comment by laq** [comment]
:+1:  :100:

**Comment by felipemontoya** [comment]
We have config vars for other cache ttls but we never change them. I think i does not make much difference

**Pull request merged**